src/gt4py/next/program_processors/runners/dace_common/dace_backend.py

```diff
@@ -26,6 +26,9 @@
 def _convert_arg(arg: Any, sdfg_param: str, use_field_canonical_representation: bool) -> Any:
     if not isinstance(arg, gtx_common.Field):
         return arg
+    if len(arg.domain.dims) == 0:
+        # pass zero-dimensional fields as scalars
+        return arg.asnumpy().item()
```

Comment:
@egparedes What was the correct pattern for avoiding type conversions?

Reply:
This type conversion is not affecting GT4Py. The GT4Py argument has still type `ts.FieldType` with empty `dims` list. This type conversion (zero-dim array to scalar) only affects the argument passed to the SDFG call, because in the SDFG I am representing the zero-dim array as a dace scalar object.

Reply:
I haven't reviewed the PR so I don't have a lot of context here, but I think Hannes' comment is about extracting the scalar value from the 0d numpy array without changing its type, because `ndarray.item()` always transforms the scalar value to a python scalar, which may change its precision. To avoid this, you should using the empty tuple as index for `.__getitem__()` like this.
```suggestion
        return arg.asnumpy()[()]
```

Reply:
Interesting, thank you.